test/src/main/java/org/apache/accumulo/test/YieldScannersIT.java

```diff
@@ -0,0 +1,106 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.accumulo.test;
+
+import org.apache.accumulo.core.client.BatchWriter;
+import org.apache.accumulo.core.client.BatchWriterConfig;
+import org.apache.accumulo.core.client.Connector;
+import org.apache.accumulo.core.client.IteratorSetting;
+import org.apache.accumulo.core.client.Scanner;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Mutation;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.accumulo.harness.AccumuloClusterHarness;
+import org.apache.accumulo.minicluster.impl.MiniAccumuloConfigImpl;
+import org.apache.accumulo.test.functional.YieldingIterator;
+import org.apache.commons.lang.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.Text;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Iterator;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// ACCUMULO-4643
+public class YieldScannersIT extends AccumuloClusterHarness {
+  Logger log = LoggerFactory.getLogger(YieldScannersIT.class);
+
+  @Override
+  public int defaultTimeoutSeconds() {
+    return 60;
+  }
+
+  @Override
+  public void configureMiniCluster(MiniAccumuloConfigImpl cfg, Configuration hadoopCoreSite) {
+    cfg.setNumTservers(1);
+  }
+
+  @Test
+  public void test() throws Exception {
+    // make a table
+    final String tableName = getUniqueNames(1)[0];
+    final Connector conn = getConnector();
+    conn.tableOperations().create(tableName);
+    final BatchWriter writer = conn.createBatchWriter(tableName, new BatchWriterConfig());
+    for (int i = 0; i < 10; i++) {
+      byte[] row = new byte[] {(byte) ('a' + i)};
+      Mutation m = new Mutation(new Text(row));
+      m.put(new Text(), new Text(), new Value());
+      writer.addMutation(m);
+    }
+    writer.flush();
+    writer.close();
+
+    log.info("Creating scanner");
+    // make a scanner for a table with 10 keys
+    final Scanner scanner = conn.createScanner(tableName, Authorizations.EMPTY);
+    final IteratorSetting cfg = new IteratorSetting(100, YieldingIterator.class);
+    scanner.addScanIterator(cfg);
+
+    log.info("iterating");
+    Iterator<Map.Entry<Key,Value>> it = scanner.iterator();
+    int keyCount = 0;
+    int yieldNextCount = 0;
+    int yieldSeekCount = 0;
+    while (it.hasNext()) {
+      Map.Entry<Key,Value> next = it.next();
+      log.info(Integer.toString(keyCount) + ": Got key " + next.getKey() + " with value " + next.getValue());
+
+      // verify we got the expected key
+      char expected = (char) ('a' + keyCount);
```

Comment:
Using a constant (instead of inline) that defines why it's being used makes much more sense to me and can avoid the implicit casts between `char` and `int`. Being a bit more terse would help with clarity, but whatever if you disagree.